uqcsbot/__init__.py

```diff
@@ -24,28 +24,40 @@
 # Mitch's UQCSTesting Slack API Token. No touchie >:(
 API_TOKEN = b64decode('eG94cC0yNjA3ODI2NzQ2MTAtMjYwMzQ1MTQ0NTI5LTMyNTEyMzU5ODExNS01YjdmYjlhYzAyZWYzNDAyNTYyMTJmY2Q2YjQ1NmEyYg==').decode('utf-8')
 
+logger = logging.getLogger("uqcsbot")
 
-def is_active_bot(user_id):
+def get_user_info(user_id):
     '''
-    Returns true if the given user_id is an active bot (i.e. not deleted).
+    Returns info about a user
+
+    See https://api.slack.com/methods/users.info for the contents of info
     '''
     api_url = 'https://slack.com/api/users.info'
     response = requests.get(api_url, params={'token': API_TOKEN, 'user': user_id})
+
     if response.status_code != requests.codes['ok']:
-        return False
+        logger.error(f'Received status code {response.status.code}')
+        sys.exit(1)
 
     json_contents = json.loads(response.content)
-    user = json_contents['user']
-    return json_contents['ok'] and user['is_bot'] and not user['deleted']
+    if not json_contents['ok']:
+        logger.error(json_contents['error'])
+        sys.exit(1)
+
+    return json_contents
+
+def is_bot_active(bot_info):
+    '''
+    Returns true if the provided bot info describes an active bot (i.e. not deleted)
+    '''
+    return bot_info['ok'] and bot_info['user']['is_bot'] and not bot_info['user']['deleted']
```

Comment:
is `is_bot` present on non-bot users? I'd probably change it to `bot_info['user'].get('is_bot', False)` just to be safe